clang/test/SemaCXX/virtual-override.cpp

```diff
@@ -289,3 +289,13 @@ namespace PR8168 {
     static void foo() {} // expected-error{{'static' member function 'foo' overrides a virtual function}}
   };
 }
+
+namespace T13 {
+  struct A {
+    virtual const int *f() const; // expected-note{{overridden virtual function is here}}
+  };
+
+  struct B : A {
+    int *f() const override; // expected-error{{virtual function 'f' has a different return type ('int *') than the function it overrides (which has return type 'const int *')}}
+  };
+}
```

Comment:
can we have additional tests with various combination of `void*`, `int*` and some class type?

Reply:
Do you mean that base return type A (`void*`, `int*` or some class type) and derived returns type B != A (`void*`, `int*` or some class type)?
Or do you mean they both return the same type but it's qualified differently?

For different types we have T1 and T2.
For different qualification of class types, we have T6.
Let me know what use case you think is uncovered.

Reply:
Given that the code has a different treatment of types which are pointers and references, having a variation of `T2` with non-class types seems valuable. (we could probably add it to `T2` directly).

Same for qualification in `T6`, given that the rules are special for the class types.

Reply:
Added tests in T2 and T6, and removed T13.
Please let me know if this is not what you meant.